front/src/Phaser/Game/CameraManager.ts

```diff
@@ -0,0 +1,95 @@
+import {RESOLUTION} from "../../Enum/EnvironmentVariable";
+import {Player} from "../Player/Player";
+import {MapManagerInterface} from "./MapManager";
+
+export interface CameraManagerInterface {
+    CurrentPlayer : Player;
+    MapManager : MapManagerInterface;
+    moveCamera() : void;
+}
+
+export class CameraManager implements CameraManagerInterface{
+    Scene : Phaser.Scene;
+    Camera : Phaser.Cameras.Scene2D.Camera;
+    CurrentPlayer : Player;
+    MapManager : MapManagerInterface;
+
+    constructor(
+        Scene: Phaser.Scene,
+        Camera : Phaser.Cameras.Scene2D.Camera,
+        MapManager: MapManagerInterface,
+        CurrentPlayer: Player
+    ) {
+        this.Scene = Scene;
+        this.MapManager = MapManager;
+        this.Camera = Camera;
+        this.CurrentPlayer = CurrentPlayer;
+    }
+    /**
+     *
+     * @param x
+     * @param y
+     * @param speedMultiplier
+     */
+    private moveCameraPosition(x:number, y:number, speedMultiplier: number): void {
+        this.Camera.scrollX += speedMultiplier * 2 * x;
+        this.Camera.scrollY += speedMultiplier * 2 * y;
+    }
+
+    /**
+     *
+     */
+    moveCamera(): void {
+        //center of camera
+        let startX = ((window.innerWidth / 2) / RESOLUTION);
+        let startY = ((window.innerHeight / 2) / RESOLUTION);
+
+        //if user client on shift, camera and player speed
+        let speedMultiplier = this.MapManager.keyShift.isDown ? 5 : 1;
+
+        if (this.MapManager.keyZ.isDown || this.MapManager.keyUp.isDown) {
```

Comment:
Instead of using the keys to move the camera, what about basing the movement of the camera on the movement of the current player?

I mean, from the CurrentPlayer player position (and the Scene object), we can automatically deduce the position of the Camera.

This will be more flexible if, in the future, the server can send a new position to the player for some reason (teleporter?) => the camera will immediately follow.
Also, this will automatically adapt the camera position if the window is resized!

Reply:
Yes I love it